Common/Data/Consolidators/BaseRenkoConsolidator.cs

```diff
@@ -0,0 +1,228 @@
+        /// value is (x => x.Value) the <see cref="IBaseData.Value"/> property on <see cref="IBaseData"/></param>
+        /// <param name="volumeSelector">Extracts the volume from a data instance. The default value is null which does
+        /// not aggregate volume per bar.</param>
+        /// <param name="evenBars">When true bar open/close will be a multiple of the barSize</param>
+        protected BaseRenkoConsolidator(
+            decimal barSize,
+            Func<IBaseData, decimal> selector,
+            Func<IBaseData, decimal> volumeSelector = null,
+            bool evenBars = true
+            )
+        {
+            EpsilonCheck(barSize);
+
+            BarSize = barSize;
+            EvenBars = evenBars;
+            Selector = selector ?? (x => x.Value);
+            VolumeSelector = volumeSelector ?? (x => 0);
+
+            Type = RenkoType.Classic;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RenkoConsolidator" /> class.
+        /// </summary>
+        /// <param name="barSize">The size of each bar in units of the value produced by <paramref name="selector"/></param>
+        /// <param name="selector">Extracts the value from a data instance to be formed into a <see cref="RenkoBar"/>. The default
+        /// value is (x => x.Value) the <see cref="IBaseData.Value"/> property on <see cref="IBaseData"/></param>
+        /// <param name="volumeSelector">Extracts the volume from a data instance. The default value is null which does
+        /// not aggregate volume per bar.</param>
+        /// <param name="evenBars">When true bar open/close will be a multiple of the barSize</param>
+        public BaseRenkoConsolidator(
+            decimal barSize,
+            PyObject selector,
+            PyObject volumeSelector = null,
+            bool evenBars = true
+            )
+            : this(barSize, evenBars)
+        {
+            EpsilonCheck(barSize);
+
+            if (selector != null)
+            {
+                if (!selector.TryConvertToDelegate(out Selector))
+                {
+                    throw new ArgumentException(
+                        "Unable to convert parameter 'selector' to delegate type Func<IBaseData, decimal>");
+                }
+            }
+            else
+            {
+                Selector = (x => x.Value);
+            }
+
+            if (volumeSelector != null)
+            {
+                if (!volumeSelector.TryConvertToDelegate(out VolumeSelector))
+                {
+                    throw new ArgumentException(
+                        "Unable to convert parameter 'volumeSelector' to delegate type Func<IBaseData, decimal>");
+                }
+            }
+            else
+            {
+                VolumeSelector = (x => 0);
+            }
+        }
+
+        /// <summary>
+        /// Gets the kind of the bar
+        /// </summary>
+        public RenkoType Type { get; }
```

Comment:
nit - members above the constructor ordered by accessibility

Reply:
Ah, it seems I was following the wrong convention! I will reorder as in https://stackoverflow.com/questions/150479/order-of-items-in-classes-fields-properties-constructors-methods